### PR TITLE
Docstring patches

### PR DESCRIPTION
**Pull request by abhro**
Major changes:
- Use of ordered lists
- Addition of method headers
- More/fixed admonition blocks
- Fixed links
- Fixed fence code examples

For more details, please see the commit messages. Thank you!

### CODE REVIEW

File: src/data/datasets_synthetic.jl [hunk 1]
```diff
@@ -318,7 +317,12 @@ Make portion `s` of vector `θ` exactly 0.
 """
 sparsify!(rng, θ, s) = (θ .*= (rand(rng, length(θ)) .< s))
 
-"""Add outliers to portion s of vector."""
+"""
+    outlify!(rng, y, s)
+
+Add outliers to portion `s` of vector.
```
Comment (ablaom) [inline src/data/datasets_synthetic.jl:323]:
```suggestion
Add outliers to portion `s` of vector `y`.
```

File: src/machines.jl [hunk 2]
```diff
@@ -572,31 +572,31 @@ the true model given by `getproperty(composite, model)`. See also [`machine`](@r
...
-  changed since the last time `mach` was trained (ie, the last time
-  `mach.state` was last incremented).
+3. The `state` of some other machine on which `mach` depends has
+   changed since the last time `mach` was trained (ie, the last time
+   `mach.state` was last incremented).
 
-- (iv) The specified `rows` have changed since the last retraining and
-  `mach.model` does not have `Static` type.
+4.  The specified `rows` have changed since the last retraining and
+   `mach.model` does not have `Static` type.
```
Comment (ablaom) [inline src/machines.jl:584]: In line 661, "iv" is explicitly used in the name of a variable `condition_iv`, which was meant to correspond to condition (iv) in the docstring which is now gone. Perhaps this variable can be renamed `condition4`. I think there's just the one example of this, but please check.

**Review by ablaom — COMMENTED** [review]
Thankyou @abhro  for this valuable contribution 🙏🏾 . Just a couple of minor comments.

**Comment by abhro** [comment]
I think I've incorporated them. Thank you!

**Pull request merged**